sv_pipeline/load_data.py

```diff
@@ -0,0 +1,392 @@
+]
+CHROM_TO_XPOS_OFFSET = {chrom: (1 + i)*int(1e9) for i, chrom in enumerate(CHROMOSOMES)}
+
+ES_TYPE_MAP = {
+    int: 'integer',
+    float: 'double',
+    str: 'keyword',
+    bool: 'boolean',
+}
+ES_FIELD_TYPLE_MAP = {
+    'xpos': 'long',
+    'xstart': 'long',
+    'xstop': 'long',
+}
+ES_INDEX_TYPE = 'structural_variant'
+NUM_SHARDS = 6
+
+
+def _get_gs_samples(project_guid, file_ext, expected_header):
+    file = GS_SAMPLE_PATH.format(project_guid=project_guid, file_ext=file_ext)
+    process = subprocess.Popen(
+        'gsutil cat {}'.format(file), stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True)
+    if process.wait() != 0:
+        return None
+    header = next(process.stdout)
+    if header.strip() != expected_header:
+        raise Exception('Missing header for sample file, expected "{}" but found {}'.format(
+            expected_header, header))
+    return [line.strip().split('\t') for line in process.stdout]
+
+
+def get_sample_subset(project_guid):
+    subset = _get_gs_samples(project_guid, file_ext='ids.txt', expected_header='s')
+    if not subset:
+        raise Exception('No sample subset file found')
+    return {row[0] for row in subset}
+
+
+def get_sample_remap(project_guid):
+    remap = _get_gs_samples(project_guid, file_ext='remap.tsv', expected_header='s\tseqr_id')
+    if remap:
+        remap = {row[0]: row[1] for row in remap}
+    return remap
+
+
+def get_field_val(row, col, header_indices, format_kwargs=None):
+    index = header_indices[col]
+    if index > len(row):
+        if COL_CONFIGS[col].get('allow_missing'):
+            return None
+        raise IndexError('Column "{}" is missing from row {}'.format(col, row))
+    val = row[header_indices[col]]
+    format_func = COL_CONFIGS[col].get('format')
+    if format_func:
+        val = format_func(val, **format_kwargs) if format_kwargs else format_func(val)
+    return val
+
+
+def get_variant_id(row, header_indices):
+    return get_field_val(
+        row, VAR_NAME_COL, header_indices, format_kwargs={'call': get_field_val(row, CALL_COL, header_indices)},
+    )
+
+
+def get_parsed_column_values(row, header_indices, columns):
+    return {COL_CONFIGS[col].get('field_name', col): get_field_val(row, col, header_indices) for col in columns}
+
+
+def parse_sv_row(row, parsed_svs_by_id, header_indices, sample_id):
+    variant_id = get_variant_id(row, header_indices)
+    if variant_id not in parsed_svs_by_id:
+        parsed_svs_by_id[variant_id] = get_parsed_column_values(row, header_indices, CORE_COLUMNS)
+        parsed_svs_by_id[variant_id][COL_CONFIGS[VAR_NAME_COL]['field_name']] = variant_id
+        parsed_svs_by_id[variant_id][GENOTYPES_FIELD] = []
+
+    sample_info = get_parsed_column_values(row, header_indices, SAMPLE_COLUMNS)
+    sample_info[SAMPLE_ID_FIELD] = sample_id
+
+    sv = parsed_svs_by_id[variant_id]
+    sv[GENOTYPES_FIELD].append(sample_info)
+    # Use the largest coordinates for the merged SV
```

Comment:
maybe I'm misreading this, but wouldn't you want to show the actual coordinates of the sample (not the largest coordinates for overlapping SVs)?